libyul/Object.cpp

```diff
@@ -31,13 +31,25 @@
 
 #include <boost/algorithm/string.hpp>
 
+#include <range/v3/view/drop.hpp>
 #include <range/v3/view/transform.hpp>
 
 using namespace solidity;
 using namespace solidity::langutil;
 using namespace solidity::util;
 using namespace solidity::yul;
 
+
+Object const* Object::at(YulString _name) const
+{
+	yulAssert(
+		this->subIndexByName.count(_name),
+		"Assembly object <" + _name.str() + "> not found."
+	);
+	size_t subIndex = this->subIndexByName.at(_name);
+	return dynamic_cast<Object const*>(this->subObjects[subIndex].get());
```

Comment:
```suggestion
	size_t subIndex = subIndexByName[_name];
	return dynamic_cast<Object const*>(subObjects[subIndex].get());
```

No need for `at` now that you've asserted that the object is indeed present and produces a valid index.

Reply:
Actually - keep the `this->`, but none of the members in `Object` are prepended with `m_` to indicate that they're indeed members, which is not great; but this isn't the place to fix it.

Reply:
> No need for `at` now that you've asserted that the object is indeed present and produces a valid index.

Indeed. However, this would make the method non-const. I would prefer to keep it const, but I can make the change if necessary.

> Actually - keep the `this->`, but none of the members in `Object` are prepended with `m_` to indicate that they're indeed members, which is not great; but this isn't the place to fix it.

Yeah, I thought about fix it here, but it is better to open a new PR for that.

Reply:
> Indeed. However, this would make the method non-const. I would prefer to keep it const, but I can make the change if necessary.

Ah, you're right, didn't even think about that.

Reply:
I'd actually remove `this->`. We never use it explicitly and it was always pointed out in my PRs.

Reply:
Also, I don't think `at()` is a good name if this method ignores `Data` objects. It's easy to take it for a general accessor like `map::at()` and assume that the index is empty when `nullptr` is returned.

The fact that this happens is not even documented in its docstring.